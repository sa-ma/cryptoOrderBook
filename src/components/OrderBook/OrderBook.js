import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import CurrencyPairs from '../CurrencyPairs/CurrencyPairs';
import Orders from '../Orders';
import Loader from '../Loader';

const OrderBook = ({ loading, error }) => {
  return (
    <>
      <CurrencyPairs />
      {loading ? (
        <Loader />
      ) : error === '' ? (
        <Orders />
      ) : (
        <div className="error">Error Loading data. Refresh the page</div>
      )}
    </>
  );
};

OrderBook.propType = {
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string.isRequired
};
const mapStateToProps = ({ currencyPairs }) => ({
  loading: currencyPairs.loading,
  error: currencyPairs.error
});
export const OrderBookComponent = OrderBook;
export default connect(mapStateToProps)(OrderBook);
