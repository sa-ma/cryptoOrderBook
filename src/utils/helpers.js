export const subscribeAction = (action, channel) => ({
  event: `bts:${action}`,
  data: {
    channel: `order_book_${channel}`
  }
});

export const getCurrency = (pair) => {
  return pair.toUpperCase().match(/.{1,3}/g);
};
