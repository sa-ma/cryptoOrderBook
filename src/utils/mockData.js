export const initialState = {
  loading: false,
  currencyPairs: [],
  error: {}
};

export const getCurrencyData = {
  data: [
    {
      base_decimals: 8,
      minimum_order: '25.0 USD',
      name: 'LTC/USD',
      counter_decimals: 2,
      trading: 'Enabled',
      url_symbol: 'ltcusd',
      description: 'Litecoin / U.S. dollar'
    }
  ]
};
export const getErrorMessage = 'Request failed with status code 500';
