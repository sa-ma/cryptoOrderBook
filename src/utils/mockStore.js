import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockStore = configureMockStore([thunk]);

const makeMockStore = (state = {}) => {
  return mockStore({
    ...state
  });
};

export default makeMockStore;
