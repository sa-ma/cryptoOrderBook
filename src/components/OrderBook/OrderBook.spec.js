import React from 'react';
import { mount, shallow } from 'enzyme';
import makeMockStore from '../../utils/mockStore';
import { Provider } from 'react-redux';
import { initialState, getCurrencyData } from '../../utils/mockData';
import { OrderBookComponent } from './OrderBook';

const store = makeMockStore({});
describe('<OrderBook/> test', () => {
  it('should render component without crashing', () => {
    const component = shallow(
      <OrderBookComponent store={store} {...initialState} />
    );
    expect(component).toMatchSnapshot();
  });
  // it('should render a loading message when currency is loading', () => {
  //   const props = {
  //     ...initialState,
  //     loading: true
  //   };
  //   const component = mount(<OrderBookComponent store={store} {...props} />);
  //   console.log(component.debug());
  //   expect(component.find('div').hasClass('loader')).toBe(true);
  // });
  // it('should render error message if no content', () => {
  //   const props = {
  //     loading: false,
  //     error: 'error'
  //   };
  //   const component = mount(<OrderBookComponent store={store} {...props} />);
  //   expect(component.find('div').hasClass('error')).toBe(true);
  // });
  // it('should render a list of currencies after loading successfully', () => {
  //   const props = {
  //     currencyPairs: [getCurrencyData],
  //     fetchCurrencyPair: jest.fn()
  //   };
  //   const component = mount(
  //     <OrderBookComponent store={store} {...props} />
  //   );
  //   expect(component.find('div').hasClass('pair-header')).toBe(true);
  //   expect(component.find('select')).toHaveLength(1);
  //   expect(component.find('label')).toHaveLength(1);
  // });
});
