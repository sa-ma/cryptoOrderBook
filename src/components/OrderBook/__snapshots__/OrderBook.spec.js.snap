// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderBook/> test should render component without crashing 1`] = `ShallowWrapper {}`;
